src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -2093,6 +2093,35 @@ public virtual void SetState(IndexState state, bool inMemoryOnly = false, bool i
                 if (State == state)
                     return;
 
+                using (DocumentDatabase.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
+                using (ctx.OpenReadTransaction())
+                using (var rawRecord = DocumentDatabase.ServerStore.Cluster.ReadRawDatabaseRecord(ctx, DocumentDatabase.Name))
+                {
+                    //If state is different then staticDef.State/autoDef.State this is not a cluster command.
+                    //In this case we can change the state only if the index state in the cluster is normal or idle
+                    if (TryFindIndexDefinition(Definition.Name, rawRecord, out IndexDefinition staticDef, out AutoIndexDefinition autoDef))
+                    {
+                        if (staticDef != null)
```

Comment:
Would it be the simpler : 
```
if(staticDef?.State == IndexState.Disabled)
   return; // log as well
```

I don't think that `Error` is a valid state for the definition..
@ppekrol Do we want to allow to create an index with an errored state?